docs/source/bundles.rst

```diff
@@ -0,0 +1,275 @@
+some custom bundle command and then write the data to a standard location that
+zipline can find. By default this location is ``$ZIPLINE_ROOT/data/<bundle>``
+where by default ``ZIPLINE_ROOT=~/.zipline``. This step may take some time as it
+could involve downloading and processing a lot of data. This can be run with:
+
+.. code-block:: bash
+
+   $ python -m zipline ingest <bundle>
+
+
+where ``<bundle>`` is the name of the bundle to ingest.
+
+Old Data
+~~~~~~~~
+
+When the ``ingest`` command is used it will write the new data to a subdirectory
+of ``$ZIPLINE_ROOT/data/<bundle>`` which is named with the current date. This
+makes it possible to look at older data or even run backtests with this older
+copy. This makers it easier to reproduce backtest results later.
+
+One drawback of saving all of this data by default is that the data directory
+may grow quite large even if you do not want to use the data. To solve this
+problem there is another command ``clean`` which will clear data bundles based
+on some time constraints.
+
+For example:
+
+.. code-block:: bash
+
+   # clean everything older than <date>
+   $ python -m zipline clean <bundle> --before <date>
+
+   # clean everything newer than <date>
+   $ python -m zipline clean <bundle> --after <date>
+
+   # keep everything in the range of [before, after] and delete the rest
+   $ python -m zipline clean <bundle> --before <date> --after <after>
+
+   # clean all but the last <int> runs
+   $ python -m zipline clean <bundle> --keep-last <int>
+
+
+Running Backtests with Data Bundles
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that the data has been ingested we can use it to run backtests with the
+``run`` command. This can be specified with the ``--bundle`` option like:
+
+.. code-block:: bash
+
+   $ python -m zipline run --bundle <bundle> --algofile algo.py ...
+
+
+We may also specify the date to use to look up the bundle data with the
+``--bundle-date`` option. This will cause us to the the most recent bundle
+ingestion that is less than or equal to the ``bundle-date``. This is how we can
+run backtests with older data. The reason that this uses a less than or equal to
+relationship is that we can specify the date that we ran an old backtest and get
+the same data that would have been available to us on that date. the
+``bundle-date`` defaults to the current day to use the most recent data.
+
+Default Data Bundles
+~~~~~~~~~~~~~~~~~~~~
+
+.. _quandl-data-bundle:
+
+Quandl WIKI Bundle
+``````````````````
+
+By default zipline comes with the ``quandl`` data bundle which uses quandl's
+`WIKI dataset <https://www.quandl.com/data/WIKI>`_. This bundle includes daily
+pricing data, splits, cash dividends, and asset metadata. This is the bundle
+that ``run`` will use by default if no other bundle is specified. To ingest this
+data bundle we recommend creating an account on quandl.com to get an API key to
+be able to make more API requests. Once we have an API key we may run:
```

Comment:
That makes more sense to me.

On Fri, Apr 29, 2016 at 4:12 PM, Joe Jevnik <notifications@github.com>
wrote:

> In docs/source/bundles.rst
> <https://github.com/quantopian/zipline/pull/1173#discussion_r61637980>:
>
> > +``bundle-date`` defaults to the current day to use the most recent data.
> > +
> > +Default Data Bundles
> > +~~~~~~~~~~~~~~~~~~~~
> > +
> > +.. _quandl-data-bundle:
> > +
> > +Quandl WIKI Bundle
> > +``````````````````
> > +
> > +By default zipline comes with the ``quandl`` data bundle which uses quandl's
> > +`WIKI dataset <https://www.quandl.com/data/WIKI>`_. This bundle includes daily
> > +pricing data, splits, cash dividends, and asset metadata. This is the bundle
> > +that ``run`` will use by default if no other bundle is specified. To ingest this
> > +data bundle we recommend creating an account on quandl.com to get an API key to
> > +be able to make more API requests. Once we have an API key we may run:
>
> does "more per day" work here?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/quantopian/zipline/pull/1173/files/6a4ec86f631ac2d62f74c536baaf6da6596a12a0#r61637980>
>